hadoop-tools/hadoop-azure/src/test/java/org/apache/hadoop/fs/azurebfs/ITestAzureBlobFileSystemCheckAccess.java

```diff
@@ -184,14 +217,15 @@ public void testCheckAccessForAccountWithoutNS() throws Exception {
     checkIfConfigIsSet(FS_AZURE_BLOB_FS_CHECKACCESS_TEST_CLIENT_SECRET);
     checkIfConfigIsSet(FS_AZURE_BLOB_FS_CHECKACCESS_TEST_USER_GUID);
 
-    setTestUserFs();
-
     //  When the driver does not know if the account is HNS enabled or not it
     //  makes a server call and fails
-    intercept(AccessControlException.class,
-        "\"This request is not authorized to perform this operation using "
-            + "this permission.\", 403",
-        () -> testUserFs.access(new Path("/"), FsAction.READ));
+    intercept(Exception.class, "\"This request is not authorized to perform this operation using "
+            + "this permission.\", 403", this::setTestUserFsNonHNS);
```

Comment:
This seems not to be required for checkAccess test.
Reason being, we intercept failure of fs initate with wrong cred.
Lets remove it? Kindly revert back if this needs to be there. Can be discussed.

Thanks!

Reply:
Since fs initiate has a getAcl call we are intercepting 403 during the initialization itself and then creating a new instance where we are configuring isNamespaceEnabled to false so the other checks in the test go through. This is done because earlier during access() call the getAcl was done and now since we are doing that during fs initialize we just need to intercept that getAcl throws 403.